examples/decomposition/plot_kernel_pca_approximate.py

```diff
@@ -0,0 +1,119 @@
+"""
+========================
+Kernel PCA Approximation
```

Comment:
Does this example show case the advantages of the new randomized solver? I think one of the major selling points is that it is faster.

Maybe the `benchmarks/bench_kernel_pca_solvers_time_vs_n_samples.py` can be turned into an example with one iteration show cases the advantages of `'dense'` vs `'randomized'` vs `'arpack'` with the following settings:

```py
min_n_samples, max_n_samples = 101, 3000  # min and max n_samples to try
n_samples_grid_size = 4                   # nb of positions in the grid to try
# generate the grid
n_samples_range = [min_n_samples + np.floor((x / (n_samples_grid_size - 1))
                                            * (max_n_samples - min_n_samples))
                   for x in range(0, n_samples_grid_size)]

n_components = 30      # the number of principal components we want to use
n_iter = 1              # the number of times each experiment will be repeated
include_arpack = True  # set this to True to include arpack solver (slower)
```

This runs fast enough to be an example and show cases the benefit of `'randomized'`. This will turn make this PR smaller by changing the benchmark into an example.

Reply:
I thought about this at the beginning, this certainly rings a bell. However what stopped me originally is that the output is a plot with execution times: not an "example" per se (one does not see the data input and output). And It looks far less nice with 4 points than the benchmark curve https://github.com/scikit-learn/scikit-learn/pull/12069#issuecomment-430270699 I'm afraid...

If you think that it is not a problem, I can certainly make the modification. Maybe then I'll be able to include the plot in the doc section, what do you think ?

Reply:
For the current example, I do not know if a user can get much actionable information from this example. I think of examples of "how can it help a user understand when to use a feature". Perhaps this PR does not need an example and the benchmarks already demonstrate its value?

Reply:
The two key messages are 

 - "the new solver is very fast especially when you do not require many of the available components". This message as you pointed out, is highlighted by the benchmarks currently
 - "in many cases that are slow today, this is because you have a large number of samples (1000, 10000). For such cases, you actually usually **do not** require many components, a few dozens (typically 100) is usually sufficient if the data contains human- or physics-based patterns and is not purely random. This is illustrated with the current example. 

Maybe we should simply merge the two in the example, by adding the 4-points simplified benchmark as you suggested ? But then I'm afraid that this could become a long example... :(

Reply:
> For such cases, you actually usually do not require many components, a few dozens (typically 100) is usually sufficient if the data contains human- or physics-based patterns and is not purely random. This is illustrated with the current example.

I agree that this is what the example shows. I am thinking that this example is showcasing something that is tangentially related to the new randomized svd feature. I think it can be its own standalone PR. (I also see that this PR greatly speeds up the example)

To try to move this PR along, this example can be part of a follow up PR.

Reply:
> I also see that this PR greatly speeds up the example

This is the point yes :)

> To try to move this PR along, this example can be part of a follow up PR.

I'll do whatever is needed to move this PR along so yes, I'll do this now !

Reply:
What about the benchmark then @thomasjpfan  ? Shall I create an example from it or leave it as is ?